<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PreferencesDialog</class>
 <widget class="QDialog" name="PreferencesDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>544</width>
    <height>595</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Preferences</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="sizeConstraint">
    <enum>QLayout::SetMinimumSize</enum>
   </property>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="generalTab">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="saveAndLoadGroupBox">
         <property name="title">
          <string>Saving and Loading</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="2" column="0">
           <widget class="QCheckBox" name="safeSaving">
            <property name="toolTip">
             <string>Turn this off if you're having trouble saving your files.</string>
            </property>
            <property name="text">
             <string>Use safe writing of files</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QCheckBox" name="restoreSession">
            <property name="text">
             <string>Restore previous session on startup</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="2">
           <widget class="QCheckBox" name="reloadTilesetImages">
            <property name="text">
             <string>&amp;Reload tileset images when they change</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QCheckBox" name="exportOnSave">
            <property name="text">
             <string>Repeat last export on save</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="exportGroupBox">
         <property name="title">
          <string>Export Options</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="2" column="0">
           <widget class="QCheckBox" name="resolveObjectTypesAndProperties">
            <property name="text">
             <string>Resolve object types and properties</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="2">
           <widget class="QCheckBox" name="embedTilesets">
            <property name="text">
             <string>Embed tilesets</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QCheckBox" name="detachTemplateInstances">
            <property name="text">
             <string>Detach templates</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QCheckBox" name="minimizeOutput">
            <property name="toolTip">
             <string>Omits unnecessary whitespace when supported by the output format</string>
            </property>
            <property name="text">
             <string>Minimize output</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="crashReporting">
         <property name="title">
          <string>Crash Reporting</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <property name="spacing">
           <number>0</number>
          </property>
          <item>
           <widget class="QCheckBox" name="sendCrashReports">
            <property name="text">
             <string>Enable sending anonymous crash reports</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="crashReportingLabel">
            <property name="text">
             <string>(&lt;a href=&quot;https://www.mapeditor.org/crash-reporting&quot;&gt;more information&lt;/a&gt;)</string>
            </property>
            <property name="textFormat">
             <enum>Qt::RichText</enum>
            </property>
            <property name="buddy">
             <cstring>sendCrashReports</cstring>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>1</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="interfaceTab">
      <attribute name="title">
       <string>Interface</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <widget class="QGroupBox" name="interfaceGroupBox">
         <property name="title">
          <string>Interface</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="5" column="1">
           <widget class="QDoubleSpinBox" name="objectLineWidth">
            <property name="suffix">
             <string> pixels</string>
            </property>
            <property name="decimals">
             <number>1</number>
            </property>
            <property name="minimum">
             <double>1.000000000000000</double>
            </property>
            <property name="value">
             <double>2.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="4" column="3">
           <spacer name="horizontalSpacer_5">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>138</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="6" column="1">
           <widget class="QComboBox" name="objectSelectionBehaviorCombo"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Grid color:</string>
            </property>
            <property name="buddy">
             <cstring>gridColor</cstring>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Background fade color:</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1" colspan="2">
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QSpinBox" name="gridMajorX">
              <property name="suffix">
               <string> tiles</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_9">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>by</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="gridMajorY">
              <property name="suffix">
               <string> tiles</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="gridFine">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>128</number>
            </property>
           </widget>
          </item>
          <item row="11" column="0">
           <widget class="QCheckBox" name="smoothScrolling">
            <property name="text">
             <string>Use s&amp;mooth scrolling</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Major grid:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="Tiled::ColorButton" name="gridColor"/>
          </item>
          <item row="10" column="0" colspan="2">
           <widget class="QCheckBox" name="autoScrolling">
            <property name="text">
             <string>Middle mouse button uses auto-&amp;scrolling</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>&amp;Language:</string>
            </property>
            <property name="buddy">
             <cstring>languageCombo</cstring>
            </property>
           </widget>
          </item>
          <item row="0" column="2" colspan="2">
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>0</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="languageCombo"/>
          </item>
          <item row="9" column="0" colspan="2">
           <widget class="QCheckBox" name="wheelZoomsByDefault">
            <property name="text">
             <string>Mouse wheel &amp;zooms by default</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="Tiled::ColorButton" name="backgroundFadeColor"/>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Object selection behavior:</string>
            </property>
            <property name="buddy">
             <cstring>objectSelectionBehaviorCombo</cstring>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Fine grid divisions:</string>
            </property>
            <property name="buddy">
             <cstring>gridFine</cstring>
            </property>
           </widget>
          </item>
          <item row="8" column="0" colspan="2">
           <widget class="QCheckBox" name="openGL">
            <property name="text">
             <string>Hardware &amp;accelerated drawing (OpenGL)</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Object line width:</string>
            </property>
            <property name="buddy">
             <cstring>objectLineWidth</cstring>
            </property>
           </widget>
          </item>
          <item row="7" column="0" colspan="2">
           <widget class="QCheckBox" name="preciseTileObjectSelection">
            <property name="text">
             <string>Pixel-perfect tile object selection</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="updatesGroupBox">
         <property name="title">
          <string>Updates</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QCheckBox" name="displayNewsCheckBox">
            <property name="text">
             <string>Display news in status bar</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="displayNewVersionCheckBox">
            <property name="text">
             <string>Highlight new version in status bar</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="Tiled::ShortcutSettingsPage" name="keyboardTab">
      <attribute name="title">
       <string>Keyboard</string>
      </attribute>
     </widget>
     <widget class="QWidget" name="updatesTab">
      <attribute name="title">
       <string>Theme</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QGroupBox" name="themeGroupBox">
         <property name="title">
          <string>Theme</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="1">
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>100</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="2">
           <widget class="QComboBox" name="styleCombo"/>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QLabel" name="selectionColorLabel">
            <property name="text">
             <string>Selection color:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="Tiled::ColorButton" name="selectionColor"/>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Style:</string>
            </property>
            <property name="buddy">
             <cstring>styleCombo</cstring>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="Tiled::ColorButton" name="baseColor"/>
          </item>
          <item row="0" column="3">
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>72</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QLabel" name="baseColorLabel">
            <property name="text">
             <string>Base color:</string>
            </property>
            <property name="buddy">
             <cstring>baseColor</cstring>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="fontGroupBox">
         <property name="title">
          <string>Use custom interface font</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="label_10">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Font:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFontComboBox" name="fontComboBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="fontSize">
            <property name="minimum">
             <number>6</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="pluginsTab">
      <attribute name="title">
       <string>Plugins</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>Enabled Plugins</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QListView" name="pluginList"/>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Extensions</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="extensionsPathLabel">
            <property name="text">
             <string>Directory:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="extensionsPathEdit">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="openExtensionsPathButton">
            <property name="text">
             <string>Open...</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Close</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Tiled::ColorButton</class>
   <extends>QToolButton</extends>
   <header>colorbutton.h</header>
  </customwidget>
  <customwidget>
   <class>Tiled::ShortcutSettingsPage</class>
   <extends>QWidget</extends>
   <header>shortcutsettingspage.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>reloadTilesetImages</tabstop>
  <tabstop>restoreSession</tabstop>
  <tabstop>safeSaving</tabstop>
  <tabstop>exportOnSave</tabstop>
  <tabstop>embedTilesets</tabstop>
  <tabstop>detachTemplateInstances</tabstop>
  <tabstop>resolveObjectTypesAndProperties</tabstop>
  <tabstop>minimizeOutput</tabstop>
  <tabstop>sendCrashReports</tabstop>
  <tabstop>languageCombo</tabstop>
  <tabstop>gridColor</tabstop>
  <tabstop>backgroundFadeColor</tabstop>
  <tabstop>gridFine</tabstop>
  <tabstop>gridMajorX</tabstop>
  <tabstop>gridMajorY</tabstop>
  <tabstop>objectLineWidth</tabstop>
  <tabstop>objectSelectionBehaviorCombo</tabstop>
  <tabstop>preciseTileObjectSelection</tabstop>
  <tabstop>openGL</tabstop>
  <tabstop>wheelZoomsByDefault</tabstop>
  <tabstop>autoScrolling</tabstop>
  <tabstop>smoothScrolling</tabstop>
  <tabstop>displayNewsCheckBox</tabstop>
  <tabstop>displayNewVersionCheckBox</tabstop>
  <tabstop>styleCombo</tabstop>
  <tabstop>baseColor</tabstop>
  <tabstop>selectionColor</tabstop>
  <tabstop>pluginList</tabstop>
  <tabstop>extensionsPathEdit</tabstop>
  <tabstop>openExtensionsPathButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>PreferencesDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>322</x>
     <y>416</y>
    </hint>
    <hint type="destinationlabel">
     <x>315</x>
     <y>312</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
